src/librustdoc/clean/mod.rs

```diff
@@ -1127,13 +1143,30 @@ impl<'a, 'tcx> Clean<Generics> for (&'a ty::Generics,
         // and instead see `where T: Foo + Bar + Sized + 'a`
 
         Generics {
-            type_params: simplify::ty_params(stripped_typarams),
-            lifetimes: gens.regions.clean(cx),
+            params: gens.regions
+                .clean(cx)
+                .into_iter()
+                .map(|lp| GenericParam::Lifetime(lp))
+                .chain(
+                    simplify::ty_params(stripped_typarams)
+                        .into_iter()
+                        .map(|tp| GenericParam::Type(tp))
+                )
+                .collect(),
             where_predicates: simplify::where_clauses(cx, where_predicates),
         }
     }
 }
 
+impl Default for Generics {
```

Comment:
You can just derive `Default` instead of manually implementing it.

Reply:
Huh, dunno why I missed that this is the same thing that the derive would generate.